src/common/ElementHandle.ts

```diff
@@ -140,6 +140,8 @@ export class ElementHandle<
   }
 
   /**
+   * @deprecated Use {@link waitForSelector} with an xpath selector.
```

Comment:
Let's update waitForSelector docs to mention that one can send xpath selectors directly.

Reply:
Added https://github.com/puppeteer/puppeteer/issues/8740. The current `waitForSelector` is insufficient anyway. Let's deal with it in another PR.